### PR TITLE
Explorations/Secrets management with xcconfig files

### LINKED ISSUE #203: Secrets manangement using xcconfig files
Based on the Xcode Cloud explorations and a discussion with @kstoletniy [on Slack](https://rootstrap.slack.com/archives/C4PFZNH9Q/p1675698160359879), we are willing to test a new approach for [Secrets management](https://www.notion.so/rootstrap/Secrets-management-d97588de41b545d19a571bfd85ba1873).

This is using `xcconfig` files, and custom scripts which are compatible with Xcode local, Xcode Cloud, and other CI/CD platforms.

More information on [xcconfig files](https://betterprogramming.pub/manage-secrets-in-your-ios-app-using-xcode-configuration-files-fbceb6e97f47), [Xcode Cloud custom env variables](https://developer.apple.com/documentation/xcode/xcode-cloud-workflow-reference#Custom-environment-variables) and, [custom scripts on Xcode Cloud workflows](https://developer.apple.com/documentation/xcode/writing-custom-build-scripts).

### PR DESCRIPTION
**Pull request by glm4**
#### Description:

* This is a set of changes proposed for the management of secrets in our apps.

The solution takes the system environment into the Info.plist of the app's target to be consumed later from the `Secret` enum.

The process consists on several steps:
1. The environment variables are available in our system's shell.
 - This is done automatically by our CI/CD platforms after you configure the env vars correctly.
 - Locally this can be done with `export ENV_VAR=value`. This is optional for local development, you could use a pre-filled `secrets.dev.xcconfig`
2. The `keys.env` file contains all the environment keys we need for the project.
3. The `setup-env` script takes the env vars from the system and write them to the secrets.xcconfig file.
 - The script takes two parameters, one is the source of the keys required(keys.env as default) and the other one is the output xcconfig file.
 - The `setup-env.sh` is configured with the default parameters for ios-base for convenience.
 - **The `secrets.xcconfig` file is committed empty to the repository and must continue that way. It is a requirement for the build to success. After that is added to the .gitignore and will be written only by the script in the CI/CD**
4.  Once the `secrets.xcconfig`  file is filled(or `secrets.dev.xcconfig` for debug), the new key can be added to the Info.plist of the app's target.
5. The Secret.Key enum represents the env keys used in the application.
6. You use like: `try? Secret.value(for: .facebookKey)` in the code.

This is a first version of the secrets management, ideally all this is moved into its own project along with init.swift

Resolves #203 
---

#### Risk:

* Medium

---

### CODE REVIEW

File: setup-env.swift [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+  private let fileManager = FileManager.default
+
+  public init(keysSource: String? = nil) {
+    self.keysSource = (keysSource ?? Defaults.keysFile).expandingTildeInPath
+  }
+
+  public func injectEnvironment(into file: String? = nil) throws {
+    let keys = try readKeys()
+
+    let fullPath = (file ?? Defaults.secretsFile).expandingTildeInPath
```
Comment (kstoletniy) [inline setup-env.swift:56]:
```suggestion
  public func injectEnvironment(into file: String = Defaults.secretsFile) throws {
    let keys = try readKeys()

    let fullPath = file.expandingTildeInPath
```

Comment (glm4) [reply setup-env.swift:56]: The `Defaults` enum is intended to be private, so can't be used as a parameter value in the signature. This approach is also convenient for allowing nil values from the command line

File: ios-base.xcodeproj/xcshareddata/xcschemes/ios-base-develop.xcscheme [hunk 2]
```diff
@@ -1,46 +1,10 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <Scheme
    LastUpgradeVersion = "1000"
-   version = "1.7">
+   version = "1.3">
```
Comment (pMalvasio) [inline ios-base.xcodeproj/xcshareddata/xcschemes/ios-base-develop.xcscheme:4]: What happened here? 🤔

**Review by pMalvasio — APPROVED** [review]

File: ios-base.xcodeproj/xcshareddata/xcschemes/ios-base-develop.xcscheme (hunk 2 above)
Comment (glm4) [reply ios-base.xcodeproj/xcshareddata/xcschemes/ios-base-develop.xcscheme:4]: Xcode stuff, maybe the pre-build actions required a higher versions, and deleting those put the version back to 1.3.

**Review by danialepaco — APPROVED** [review]

**Review by LetoFranco — APPROVED** [review]

**Review by fpiruzi — APPROVED** [review]

**Pull request merged**